x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.ts

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { clearFlashMessages, flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { KibanaLogic } from '../../../shared/kibana';
+import { ANY_AUTH_PROVIDER } from '../../../shared/role_mapping/constants';
+import { AttributeName } from '../../../shared/types';
+import { ROLE_MAPPINGS_PATH } from '../../routes';
+import { ASRoleMapping, RoleTypes } from '../../types';
+import { roleHasScopedEngines } from '../../utils/role/has_scoped_engines';
+import { Engine } from '../engine/types';
+
+import { DELETE_ROLE_MAPPINGS_MESSAGE } from './constants';
+
+interface RoleMappingsServerDetails {
+  roleMappings: ASRoleMapping[];
+  multipleAuthProvidersConfig: boolean;
+}
+
+interface RoleMappingServerDetails {
+  attributes: string[];
+  authProviders: string[];
+  availableEngines: Engine[];
+  elasticsearchRoles: string[];
+  hasAdvancedRoles: boolean;
+  multipleAuthProvidersConfig: boolean;
+  roleMapping?: ASRoleMapping;
+}
+
+const getFirstAttributeName = (roleMapping: ASRoleMapping) =>
+  Object.entries(roleMapping.rules)[0][0] as AttributeName;
```

Comment:
[optional/not a blocker] you could also just do `Object.keys(roleMapping.rules)[0]`, might be a bit more straightforward

Reply:
Noted. Because of the decision to rewrite this later, I'm going to pass on non-blocking/optional fixes